packages/components/src/Box.js

```diff
@@ -9,10 +10,20 @@ const shouldForwardProp = createShouldForwardProp([
   ...color.propNames,
 ])
 
-const sx = props => css(props.sx)(props.theme)
-const base = props => css(props.__css)(props.theme)
-const variant = ({ theme, variant, __themeKey = 'variants' }) =>
-  css(get(theme, __themeKey + '.' + variant, get(theme, variant)))
+const sx = (props) => css(props.sx)(props.theme)
+const base = (props) => css(props.__css)(props.theme)
+const variant = ({ theme, variant, __themeKey = 'variants' }) => {
+  if (Array.isArray(variant)) {
+    return css(
+      variant.reduce(
```

Comment:
No benchmarks yet, but I have a feeling we should cache this.

Also, I don't think `deepmerge` does the job here.
Variants are just style objects, so they can contain functions -- both `Theme => ResponsiveStyleValue` and `Theme => ThemeUIStyleObject`.
Using deepmerge here feels inconsistent with regards to derived style objects (`Theme => ThemeUIStyleObject`).

Reply:
You are right - #586 is the same PR - i can try to update #586(a few things have probbaly changed), or update my PR a bit to use their deepmerge.all, or just shallow object copy.
Functions i think should be fine - i can a test

Reply:
I think if you're doing a _deep merge_ and function represent _styled object derived from the theme_ then a merged function is `(theme) => deepmerge(f(theme), g(theme))`.

Reply:
@hasparus - I update the merge/reduce to a deepmerge.all call, also added test `renders with variant array function merge`.

For the function merge - can you give me an example to include in the tests? So far I assumed the theme functions resolve to a string, not to an object.

Reply:
I'll add a test to your branch to explain what I mean, if you don't mind.

Reply:
yes please

Reply:
Here it is :)